spec/pluginmanager/sources_spec.rb

```diff
@@ -0,0 +1,150 @@
+# encoding: utf-8
+require "spec_helper"
+require 'webmock/rspec'
+require "tmpdir"
+require "pluginmanager/sources/http"
+require "pluginmanager/sources/local"
+
+describe LogStash::PluginManager::Sources::HTTP do
+
+  subject   {  described_class.new(uri) }
+
+  let(:uri) { "http://localhost:8080/pack.zip"}
+
+  describe "#exist?" do
+
+    it "should verify if a plugin exist in the repository" do
+      stub_request(:head, uri).and_return(:status => 200)
+      expect(subject.exist?).to eq(true)
+    end
+
+    it "should fail if a plugin does not exist in the repository" do
+      stub_request(:head, uri).and_return(:status => 404)
+      expect(subject.exist?).to eq(false)
+    end
+
+  end
+
+  describe "#fetch" do
+
+    before(:each) do
+      File.delete("pack.zip") if File.exist?("pack.zip")
+    end
+
+    after(:each) do
+      File.delete("pack.zip") if File.exist?("pack.zip")
+    end
+
+    it "should start the download process" do
+      stub_request(:get, uri).and_return(:body => "foo", :status => 404)
+      Dir.mktmpdir do |tmp_dir|
```

Comment:
sure, but still no technical benefit for this use case here, I must
disagree of using for using libs :-)

On Tue, Apr 19, 2016 at 3:00 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In spec/pluginmanager/sources_spec.rb
> <https://github.com/elastic/logstash/pull/4584#discussion_r60225452>:
>
> > +
> > +  end
> > +
> > +  describe "#fetch" do
> > +
> > +    before(:each) do
> > +      File.delete("pack.zip") if File.exist?("pack.zip")
> > +    end
> > +
> > +    after(:each) do
> > +      File.delete("pack.zip") if File.exist?("pack.zip")
> > +    end
> > +
> > +    it "should start the download process" do
> > +      stub_request(:get, uri).and_return(:body => "foo", :status => 404)
> > +      Dir.mktmpdir do |tmp_dir|
>
> Because its a library we maintain and we use it everywhere when dealing
> with temporary files and directory. I think consistency around this need is
> good.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4584/files/3b56017c73f4326a11c3abd6b87a0c8f7dbb811b#r60225452>
>